src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -26,11 +33,71 @@
     private final ItemModel model;
     private final Storage storage;
     private final AddressBookParser addressBookParser;
+    private final ScheduledThreadPoolExecutor checker;
 
     public LogicManager(ItemModel model, Storage storage) {
         this.storage = storage;
         this.model = model;
+
         addressBookParser = new AddressBookParser(model.getElisaStateHistory());
+
+        //Create new thread class to check
+        /*
+        class CheckerThread extends Thread {
+
+            ScheduledExecutorService
+            //Schedule a check on future list every 5 seconds
+            //If future.first.time <= now
+            // A new collection <- future.remove until future.first.time > now
+            // ActiveRemindersListProperty.add(A new collection)
+            //
+        }
+        */
+        //Spawn new thread
+        //new CheckerThread().start();
+        //Run checker
+        Runnable checkTask = new Runnable() {
+            public void run() {
+                System.out.println("running checkTask");
+                Reminder reminder;
+                ArrayList<Item> reminders = model.getFutureRemindersList();
+                ArrayList<Item> activeReminders = new ArrayList<Item> (0);
+                if (reminders.size() > 0) {
+                    System.out.println("There are pending reminders: " + reminders.toString());
+                    //TODO: Check if Optional is present before .get()
+                    reminder = reminders.get(0).getReminder().get();
+                    synchronized (reminders) {
+                        while (reminder != null && reminder.getDateTime().isBefore(LocalDateTime.now())) {
+                            System.out.println("Adding a new task to be reminded.");
+                            Item item = reminders.remove(0);
+                            activeReminders.add(item);
+                            if (reminders.size() > 0) {
+                                reminder = reminders.get(0).getReminder().get();
+                            } else {
+                                reminder = null;
+                            }
+                        }
+                    }
```

Comment:
Perhaps you can place this in another method? This can help to reduce the level of indentation and try to maintain the same level of abstraction in this method :)

Reply:
Good idea! I'll create a new issue for this.